### PR TITLE
Export test results test addon

### PR DESCRIPTION
**Pull request by lszucs**

### CODE REVIEW

**Comment by trapacska** [comment]
Please separate addon related logic to a separate package. Also you can keep android project based logic in a shared package but do not have it under the addon package.

File: main.go [hunk 1]
```diff
@@ -102,6 +109,14 @@ func filterVariants(module, variant string, variantsMap gradle.Variants) (gradle
 	return filteredVariants, nil
 }
 
+func artifactsToMap(artifacts []gradle.Artifact) map[string]string {
```
Comment (godrei) [inline main.go]: seems to be unused

File: main.go [hunk 2]
```diff
@@ -196,6 +211,28 @@ func main() {
...
+	}
+
+	if baseDir := os.Getenv("BITRISE_TEST_RESULT_DIR"); baseDir != "" {
+		for _, artifact := range resultXMLs {
+			uniqueDir, err := getUniqueDir(artifact.Path)
+			if err != nil {
+				log.Warnf("Failed to export test results for test addon: cannot get export directory for artifact (%s): %s", err)
+			}
+
+			if err := testaddon.ExportArtifact(artifact.Path, baseDir, uniqueDir); err != nil {
```
Comment (godrei) [inline main.go]: does it make sense to call this if an error occurred above?

File: testaddon.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	i := len(parts) - 1
+	for i > 0 && parts[i] != "test-results" {
+		i--
+	}
+
+	if i == 0 {
+		return "", fmt.Errorf("path (%s) does not contain 'test-results' folder", path)
+	}
+
+	return parts[i-2], nil
```
Comment (godrei) [inline testaddon.go]: just to be sure, pls validate the index before indexing the array

File: testaddon.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	i := len(parts) - 1
+	for i > 0 && parts[i] != "test-results" {
+		i--
+	}
+
+	if i == 0 {
+		return "", fmt.Errorf("path (%s) does not contain 'test-results' folder", path)
+	}
+
+	variant := parts[i+1]
```
Comment (godrei) [inline testaddon.go:25]: just to be sure, pls validate the index before indexing the array

File: testaddon.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+// getModule returns the name of the module a given test result artifact was produced by
+// based of the artifacts path.
+// path example: <PATH_TO_YOUR_PROJECT>/<MODULE_NAME>/build/test-results/testDemoDebugUnitTest/TEST-example.com.helloworld.ExampleUnitTest.xml
+func getModule(path string) (string, error) {
```
Comment (godrei) [inline testaddon.go]: pls merge with `getVariant`

File: testaddon/testaddon.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		return err
+	}
+
+	return nil
+}
+
+// ExportArtifact exports artifact found at path in directory uniqueDir,
+// rooted at baseDir.
+func ExportArtifact(path, baseDir, uniqueDir string) error {
+		exportDir := strings.Join([]string{baseDir, uniqueDir}, "/")
```
Comment (godrei) [inline testaddon/testaddon.go]: pls use filepath.Join

**Review by godrei — CHANGES_REQUESTED** [review]
pls have a look at my notes

**Review by godrei — APPROVED** [review]

**Pull request merged**